core/domain/exp_services.py

```diff
@@ -1188,7 +1322,12 @@ def update_exploration(
             'Commit messages for non-suggestions may not start with \'%s\'' %
             feconf.COMMIT_MESSAGE_ACCEPTED_SUGGESTION_PREFIX)
 
-    exploration = apply_change_list(exploration_id, change_list)
+    exploration_before_change = get_exploration_by_id(exploration_id)
```

Comment:
Why do we need old exploration here? Can't we just check which of the image ids are not used in the html and only delete those image ids?  is there any issue with that?

Reply:
I think you are right, i am doing by wrong technique
We can run one loop and check which of the ids are not exist in html.